context/app/static/js/shared-styles/inputs/MultiAutocomplete/MultiAutocomplete.tsx

```diff
@@ -0,0 +1,49 @@
+import React, { useState } from 'react';
+
+import Autocomplete, { AutocompleteProps } from '@mui/material/Autocomplete';
+import TextField, { TextFieldProps } from '@mui/material/TextField';
+import { InputLabelProps as inputLabelProps } from '@mui/material/InputLabel';
+
+interface MultiAutocompleteTypes extends AutocompleteProps {
+  tagComponent: React.ReactNode;
+  renderInputProps: TextFieldProps;
+}
+
+interface RenderInputTypes {
+  InputLabelProps: inputLabelProps;
+  textFieldProps: TextFieldProps;
+}
+
+function MultiAutocomplete({ renderInputProps, tagComponent: TagComponent, ...rest }): MultiAutocompleteTypes {
+  const [substring, setSubstring] = useState('');
+
+  function handleChange({ target: { value } }: React.ChangeEvent<HTMLInputElement>) {
+    setSubstring(value);
+  }
+
+  return (
+    <Autocomplete
+      renderTags={(value, getTagProps) =>
+        value.map((option, index) => {
+          return <TagComponent option={option} {...getTagProps({ index })} key={option} />;
+        })
+      }
+      renderInput={({ InputLabelProps, ...textFieldProps }: RenderInputTypes) => (
```

Comment:
`renderInput` has an expected param type provided by MUI (`AutocompleteRenderInputParams`) so we shouldn't need to define these types. `ctrl + click`ing on the `renderInput` prop name opens the definition from the MUI types, then `ctrl + click`ing the params type in there shows what you have access to:

```
export interface AutocompleteRenderInputParams {
  id: string;
  disabled: boolean;
  fullWidth: boolean;
  size: 'small' | undefined;
  InputLabelProps: ReturnType<ReturnType<typeof useAutocomplete>['getInputLabelProps']>;
  InputProps: {
    ref: React.Ref<any>;
    className: string;
    startAdornment: React.ReactNode;
    endAdornment: React.ReactNode;
  };
  inputProps: ReturnType<ReturnType<typeof useAutocomplete>['getInputProps']>;
}
```